### PR TITLE
C14 : init

### PR DESCRIPTION
**Pull request by apeyroux**
###### Motivation for this change

C14 is designed for data archiving & long-term backups and lets you store and retrieve large amounts of data.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/c14/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+buildGoPackage rec {
+  name = "c14-cli-${version}";
+  version = "20160909-${stdenv.lib.strings.substring 0 7 rev}";
+  rev = "e7c7c3cb214fd06df63530a4e861210e7a0a1b6c";
+
+  goPackagePath = "github.com/online-net/c14-cli";
+  subPackages = [ "cmd/c14" ];
+  
+  src = fetchgit {
```
Comment (groxxda) [inline pkgs/applications/networking/c14/default.nix]: please use `fetchFromGitHub` if possible

File: pkgs/applications/networking/c14/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{ stdenv, buildGoPackage, fetchgit, fetchhg, fetchbzr, fetchsvn }:
```
Comment (groxxda) [inline pkgs/applications/networking/c14/default.nix]: please remove the `fetch*` imports that are unused

File: pkgs/applications/networking/c14/default.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+buildGoPackage rec {
+  name = "c14-cli-${version}";
+  version = "20160909-${stdenv.lib.strings.substring 0 7 rev}";
```
Comment (groxxda) [inline pkgs/applications/networking/c14/default.nix]: date should be hyphenated (yyyy-mm-dd)

**Review by groxxda — CHANGES_REQUESTED** [review]
Please also consider adding a `meta` section

File: pkgs/applications/networking/c14/default.nix (hunk 1 above)
Comment (41px) [inline pkgs/applications/networking/c14/default.nix]: goPackagePath does not facilitate the use of packages go?

Comment (41px) [inline pkgs/applications/networking/c14/default.nix]: I had watched networking/drive before. https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/drive/default.nix

**Comment by Mic92** [comment]
I would do the following changes: 

```diff
diff --git a/pkgs/applications/networking/c14/default.nix b/pkgs/applications/networking/c14/default.nix
index a3d1c5f..3a7e1c7 100644
--- a/pkgs/applications/networking/c14/default.nix
+++ b/pkgs/applications/networking/c14/default.nix
@@ -1,16 +1,14 @@
-{ stdenv, buildGoPackage, fetchgit }:
+{ stdenv, buildGoPackage, fetchFromGitHub }:
 
 buildGoPackage rec {
-  name = "c14-cli-${version}";
-  version = "09092016-${stdenv.lib.strings.substring 0 7 rev}";
-  rev = "e7c7c3cb214fd06df63530a4e861210e7a0a1b6c";
-
+  name = "c14-cli-2016-09-09";
   goPackagePath = "github.com/online-net/c14-cli";
   subPackages = [ "cmd/c14" ];
-  
-  src = fetchgit {
-    inherit rev;
-    url = "https://github.com/online-net/c14-cli";
+
+  src = fetchFromGitHub {
+    owner = "online-net";
+    repo = "c14-cli";
+    rev = "e7c7c3cb214fd06df63530a4e861210e7a0a1b6c";
     sha256 = "1k53lii2c04j8cy1bnyfvckl9fglprxp75cmbg15lrp9iic2w22a";
   };
```

File: pkgs/applications/networking/c14/default.nix [hunk 4]
```diff
@@ -20,6 +20,5 @@ buildGoPackage rec {
     description = "C14 is designed for data archiving & long-term backups.";
     homepage = "https://www.online.net/en/c14";
     license = licenses.mit;
-    maintainers = with maintainers; [ 41px ];
```
Comment (Mic92) [inline pkgs/applications/networking/c14/default.nix]: Is there a reason, you stepped back from maintenance?

**Comment by apeyroux** [comment]
See more on #19156

**Pull request closed without merging**